core/src/test/java/org/elasticsearch/index/translog/TranslogTests.java

```diff
@@ -2178,4 +2182,137 @@ public void testRollGenerationBetweenPrepareCommitAndCommit() throws IOException
 
     }
 
+    public void testMinGenerationForSeqNo() throws IOException {
+        final int operations = randomIntBetween(1, 4096);
+        final List<Long> seqNos =
+                LongStream.range(0, operations).boxed().collect(Collectors.toList());
+        Randomness.shuffle(seqNos);
+        final Map<Long, List<Long>> generations = new HashMap<>();
+
+        for (int i = 0; i < operations; i++) {
+            final Long seqNo = seqNos.get(i);
```

Comment:
it would be great to add duplicate seq# to this test. The real engine needs to deal with multiple versions of the same seq#, distinguished only by their associated term.